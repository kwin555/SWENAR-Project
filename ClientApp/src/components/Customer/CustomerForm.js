import React, { Component } from "react";

class CustomerForm extends Component {
  //store default state of application
  constructor(props) {
    super(props);

    this.state = {
      Name: "",
      number: "",
      error: false,
    };
  }
  // call to handle the value in a text input
  onChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  //call to create a new customer
  submitToPost = () => {
    let curr = this;
    const axios = require("axios");
    const { Name, number } = this.state;
    axios
      .post("/api/customer", {
        Name: Name,
        Number: number,
      })
      .then(function (response) {
        curr.props.handleNumName(number, Name);
      })
      .catch(function (error) {
        curr.setState({ error: true });
      });
  };

  render = () => {
    return (
      <div>
        <form>
          <label htmlFor='Name'>Enter name: </label>
          <input id='Name' name='Name' type='text' onChange={this.onChange} />

          <label htmlFor='Number'>Enter customer number:</label>
          <input
            id='number'
            name='number'
            type='text'
            onChange={this.onChange}
          />
        </form>
        <div>
          <button onClick={this.submitToPost}>Add</button>
        </div>
      </div>
    );
  };
}

export default CustomerForm;
