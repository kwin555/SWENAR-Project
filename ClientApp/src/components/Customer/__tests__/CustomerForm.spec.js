import React from "react";
import CustomerForm from "../CustomerForm";
import { shallow } from "enzyme";

describe("CustomerForm tests", () => {
  it("shows customer forms with things rendered", () => {
    const props = {
      Name: "",
      number: "",
    };
    const wrapper = shallow(<CustomerForm {...props}></CustomerForm>);
    expect(wrapper).toMatchSnapshot();
  });
});
