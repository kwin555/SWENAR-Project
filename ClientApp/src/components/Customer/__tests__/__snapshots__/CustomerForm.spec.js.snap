// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerForm tests shows customer forms with things rendered 1`] = `
<div>
  <form>
    <label
      htmlFor="Name"
    >
      Enter name: 
    </label>
    <input
      id="Name"
      name="Name"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="Number"
    >
      Enter customer number:
    </label>
    <input
      id="number"
      name="number"
      onChange={[Function]}
      type="text"
    />
  </form>
  <div>
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
